src/services/core.service.ts

```diff
@@ -159,25 +173,83 @@ export class CoreService {
             .concat('/products/' + this.config.productID)
             .concat('/versions/' + this.config.version)
             .concat('/locales/' + language)
-            .concat('/components/' + this.config.component)
+            .concat('/components/' + component)
             .concat('?pseudo=' + this.config.isPseudo);
     }
+
+    /**
+     * Get translation request url by multiple components
+     * @param component
+     * @param language
+     */
+    private getMultiComponentTransUrl(language: string, component: string) {
+        // todo: support i18nAssets
+        return this.config.host
+        .concat(Constants.L10N_COMPONENT_API_ENDPOINT)
+        .concat('/products/' + this.config.productID)
+        .concat('/versions/' + this.config.version)
+        .concat('?components=' + component)
+        .concat('&locales=' + language)
+        .concat('&pseudo=' + this.config.isPseudo);
+    }
+
     /**
      * return promise with processed result.
      * @param language
      */
-    public loadTranslations(language: string): Promise<any> {
-        const cache = this.cacheManager.lookforTranslationByComponent(this.getComponent(), language);
+    public loadTranslations(language: string): Promise<any> | Promise<any>[] {
+        const isMultipleComponents = this.config.components && this.config.components.length > 0;
+        const isCombine = isMultipleComponents && this.config.combineRequest;
+
```

Comment:
whether need to logic judge for isMultipleComponents = false and combineRequest = ture when wrong configuration during initilizing VIP